interface/src/app/gene/resource.js

```diff
@@ -1,22 +1,29 @@
-angular.module('adage.gene.resource', ['ngResource'])
+angular.module('adage.gene.resource', [
+  'ngResource',
+  'adage.utils'
+])
 
-.factory('Gene', ['$resource', function($resource) {
+.factory('Gene', ['$resource', 'ApiBasePath', function($resource, ApiBasePath) {
   return $resource(
-    '/api/v0/gene/:id', {
+    ApiBasePath + 'gene/:id', {
       id: '@id'
     }, {
-
+      post: {
+        url: ApiBasePath + 'gene/',
```

Comment:
I just saw the part of your comment that says "if we do not send the `id` as a parameter..."

You are absolutely right. I guess what I'm doing is protecting against future changes to the code that might go and add that parameter in. I'd rather have the Gene service keep working properly  under that condition rather than leave this issue to be discovered and fixed later.